amazon/ion/ioncmodule.c

```diff
@@ -1037,6 +1009,34 @@ static iERR ionc_read_into_container(hREADER hreader, PyObject* container, BOOL
     iRETURN;
 }
 
+/*
+ *  Adds an element to a List or struct
+ *
+ *  Args:
+ *      pyContainer:  A container that the element is added to
+ *      element:  The element to be added to the container
+ *      in_struct:  if the current state is in a struct
+ *      field_name:  The field name of the element if it is inside a struct
+ */
+static void ionc_add_to_container(PyObject* pyContainer, PyObject* element, BOOL in_struct, ION_STRING* field_name) {
```

Comment:
Slight apologies for moving and changing, but it was super annoying that it was 1000 lines above the one place that called it!